node/cli/src/factory_impl.rs

```diff
@@ -51,6 +51,16 @@ pub struct FactoryState<N> {
 
 type Number = <<node_primitives::Block as BlockT>::Header as HeaderT>::Number;
 
+impl<Number> FactoryState<Number> {
+	fn build_extra(index: node_primitives::Index, phase: u64) -> node_runtime::SignedExtra<Runtime> {
```

Comment:
@gavofyork how about this? The factory seems to be an attachment that only works with substrate node and it should be fine to borrow this type.